search/src/main/scala/ch.epfl.scala.index.search/DataRepository.scala

```diff
@@ -95,85 +148,92 @@ class DataRepository(esClient: TcpClient, indexName: String)(implicit
   def updateProject(project: Project): Future[Unit] = {
     esClient
       .execute(
-        update(project.id.get)
-          .in(indexName / projectsCollection)
-          .doc(project)
+        updateById(projectIndex, project.id.get).doc(project)
       )
       .map(_ => ())
   }
 
-  def insertReleases(releases: Seq[Release]): Future[RichBulkResponse] = {
+  def insertReleases(releases: Seq[Release]): Future[BulkResponse] = {
     val requests = releases.map { r =>
-      indexInto(indexName / releasesCollection).source(ReleaseDocument(r))
+      indexInto(releaseIndex).source(ReleaseDocument(r))
     }
 
-    esClient.execute(bulk(requests))
+    esClient.execute(bulk(requests)).map(_.result)
   }
 
   def insertRelease(release: Release): Future[Unit] = {
     esClient
       .execute {
-        indexInto(indexName / releasesCollection)
+        indexInto(releaseIndex)
           .source(ReleaseDocument(release))
       }
       .map(_ => ())
   }
 
   def insertDependencies(
       dependencies: Seq[ScalaDependency]
-  ): Future[RichBulkResponse] = {
+  ): Future[Seq[BulkResponseItem]] = {
     if (dependencies.nonEmpty) {
       val requests = dependencies.map { d =>
-        indexInto(indexName / dependenciesCollection)
+        indexInto(dependencyIndex)
           .source(DependencyDocument(d))
       }
-      esClient.execute(bulk(requests))
+      Future
+        .sequence(
+          requests
+            .grouped(1000)
```

Comment:
Do you to do this manually?

Reply:
I had to chunk this request because of the size limit imposed by the default configuration of elasticsearch 7

Reply:
I understand. What I do not like is that *you* have to do it *manually* and here. Isn't `bulk` meant to address this problem?

Reply:
`bulk` is an HTTP endpoint to insert many document in a single request. So the logic is not in the library but in the elasticsearch server.